lincheck/src/main/java/org/jetbrains/kotlinx/lincheck/strategy/ManagedStrategyTransformer.java

```diff
@@ -48,80 +46,659 @@
+        this.codeLocations = codeLocations;
+        this.classVersion = ci.getVersion();
         this.fileName = ci.getSourceFile();
     }
 
-    public List<StackTraceElement> getCodeLocations() {
+    List<StackTraceElement> getCodeLocations() {
         return codeLocations;
     }
 
     @Override
     public MethodVisitor visitMethod(int access, String mname, String desc, String signature, String[] exceptions) {
+        //System.out.println(className);
+        // replace native method VMSupportsCS8 in AtomicLong with stub
+        if ((access & ACC_NATIVE) != 0 && mname.equals("VMSupportsCS8")) {
+            MethodVisitor mv = super.visitMethod(access & ~ACC_NATIVE, mname, desc, signature, exceptions);
+            return new NativeMethodStubTransformer(new GeneratorAdapter(mv, access & ~ACC_NATIVE, mname, desc));
+        }
+
+        boolean isSynchronized = (access & ACC_SYNCHRONIZED) != 0;
+        if (isSynchronized) {
+            access ^= ACC_SYNCHRONIZED; // disable synchronized
+        }
+
         MethodVisitor mv = super.visitMethod(access, mname, desc, signature, exceptions);
         mv = new JSRInlinerAdapter(mv, access, mname, desc, signature, exceptions);
+        mv = new IndirectSharedVariableAccessMethodTransformer(mname, new GeneratorAdapter(mv, access, mname, desc));
+        mv = new SynchronizedLockTransformer(mname, new GeneratorAdapter(mv, access, mname, desc));
+
+        if (isSynchronized) {
+            // synchronized method is replaced with synchronized lock
+            mv = new SynchronizedLockAddingTransformer(mname, new GeneratorAdapter(mv, access, mname, desc), className, access, classVersion);
+        }
+
+        if ("<clinit>".equals(mname)) {
+            mv = new ClassInitializationTransformer(mname, new GeneratorAdapter(mv, access, mname, desc));
+        }
+
+        mv = new WaitNotifyTransformer(mname, new GeneratorAdapter(mv, access, mname, desc));
+        mv = new ParkUnparkTransformer(mname, new GeneratorAdapter(mv, access, mname, desc));
+        mv = new UnsafeTransformer(new GeneratorAdapter(mv, access, mname, desc));
+        // SharedVariableAccessMethodTransformer should be an earlier visitor than ClassInitializationTransformer
+        // not to have suspension points before 'beforeClassInitialization' call in <clinit> block.
+        mv = new LocalObjectManagingTransformer(mname, new GeneratorAdapter(mv, access, mname, desc));
         mv = new SharedVariableAccessMethodTransformer(mname, new GeneratorAdapter(mv, access, mname, desc));
-        // TODO add other transformers
         mv = new TryCatchBlockSorter(mv, access, mname, desc, signature, exceptions);
         return mv;
     }
 
-    class SharedVariableAccessMethodTransformer extends ManagedStrategyMethodVisitor {
-        public SharedVariableAccessMethodTransformer(String methodName, GeneratorAdapter mv) {
+    /**
+     * Changes package of some transformed classes, because they cannot stay in the same package
```

Comment:
The remapper name specifies these "some" classes, change the comment correspondingly, please.